src/modules/Dashboard/features/Participant/Assignments/ByParticipant/ByParticipant.tsx

```diff
@@ -29,35 +40,86 @@ const ByParticipant = () => {
 
-  const handleRefetch = useCallback(() => {
+  const handleRefetchExpandedView = useCallback(
+    async (activityOrFlowId: string) => {
+      if (!respondentSubjectId) return;
+
+      setExpandedViewsLoading((prev) => ({ ...prev, [activityOrFlowId]: true }));
+
+      try {
+        const { data } = await getTargetSubjectsByRespondentApi({
+          activityOrFlowId,
+          subjectId: respondentSubjectId,
+        });
+        // Add or update the correspondent element in expandedViewsData
+        setExpandedViewsData((prev) => ({ ...prev, [activityOrFlowId]: data.result }));
+      } finally {
+        setExpandedViewsLoading((prev) => ({ ...prev, [activityOrFlowId]: false }));
+      }
+    },
+    [respondentSubjectId],
+  );
+
+  const handleRefetchActivities = useCallback(() => {
     // Avoid fetching activities for respondent if respondent is a limited account
-    if (!appletId || !subjectId || !respondentSubject?.userId) return;
+    if (!appletId || !respondentSubject?.id || !respondentSubject.userId) return;
+
+    fetchActivities({ appletId, subjectId: respondentSubject.id });
+  }, [appletId, fetchActivities, respondentSubject]);
+
+  const handleRefetchAll = useCallback(() => {
+    handleRefetchActivities();
 
-    fetchActivities({ appletId, subjectId });
-  }, [appletId, fetchActivities, respondentSubject?.userId, subjectId]);
+    // Refresh target subject data for any expanded views
+    Object.entries(expandedViewsData).forEach(([id, viewData]) => {
+      if (viewData) void handleRefetchExpandedView(id);
+    });
+  }, [handleRefetchActivities, expandedViewsData, handleRefetchExpandedView]);
 
   const {
     getActionsMenu,
     onClickAssign,
+    onClickNavigateToData,
     isLoading: isLoadingHook,
     modals,
-  } = useAssignmentsTab({ appletId, respondentSubject, handleRefetch, dataTestId });
-
-  /*
-  TODO: Handler for navigating to data when card is expanded
-  https://mindlogger.atlassian.net/browse/M2-7921
-  const handleClickNavigateToData = (activityOrFlow: ParticipantActivityOrFlow, targetSubject: RespondentDetails) => {
+  } = useAssignmentsTab({
+    appletId,
+    respondentSubject,
+    handleRefetch: handleRefetchAll,
+    dataTestId,
+  });
+
+  const handleClickNavigateToData = (
+    activityOrFlow: ParticipantActivityOrFlow,
+    targetSubject: RespondentDetails,
+  ) => {
     if (!respondentSubject) return;
 
     onClickNavigateToData(activityOrFlow, targetSubject.id);
   };
-  */
+
+  const handleClickToggleExpandedView = useCallback(
+    (isExpanded: boolean, activityOrFlowId: string) => {
+      if (!respondentSubject) return;
+
+      if (isExpanded) {
+        void handleRefetchExpandedView(activityOrFlowId);
+      } else {
+        // If expanded view is closed, remove data to free up memory and minimize refetches
+        // (after delay to account for transition)
+        setTimeout(() => {
+          setExpandedViewsData((prev) => ({ ...prev, [activityOrFlowId]: undefined }));
```

Comment:
Somehow your comment got lost @sultanofcardio, I guess during rebase? Just wanted to respond:

> Setting the entry to undefined will still keep it in the object, and later on Object.entries.forEach will still execute a re-fetch for it. Remove it instead

I accounted for `undefined` entries [above](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1943/files#diff-5fb2ade196f939ddef9b4ecea13b66a6e82119397515bbbb236193b54dd63cd5R75), only refetching if data is present for that entry. I considered your suggested approach and didn't like that the `const { …, ...rest } = …` approach took up 4 extra lines in the file (lame, yes 😛). But it would mean one less conditional during refetch, so I'll do that.

Reply:
It didn't get lost, I later realised that it was accounted for so I deleted the comment. I should've probably replied to it instead 😅

Reply:
I'm just as sneaky, rewording my reply to you like 5 times before I was satisfied lol… 🙌